### PR TITLE
[slonik] Update createPool() and add missing errors

### LINKED ISSUE #157: Wrong typings for createPool arguments
The types currently specify that createPool's first argument can be either a string or an object that describes the postgres connection.

## Expected Behavior

Create pool should accept either a string **or** a configuration object as specified in the types here:
```ts
export function createPool(
    connectionConfiguration: DatabaseConfigurationType,
    clientUserConfiguration?: ClientConfigurationInputType
): DatabasePoolType;

...

export type DatabaseConfigurationType =
    | string
    | {
        database?: string;
        host?: string;
        idleTimeoutMillis?: number;
        max?: number;
        password?: string;
        port?: number;
        user?: string;
    };
```

## Current Behavior
createPool only accepts strings

https://github.com/gajus/slonik/blob/master/src/factories/createPool.js#L22

## Possible Solution
Not sure in this case, confused why the types would reflect a feature that doesn't exist. Is it possible to accept a valid pool configuration just like in `node-postgres`

## Steps to Reproduce
```ts
const pool = createPool({host: 'localhost:5432', etc...})
```

## Logs
```
  if(str.charAt(0) === '/') {
         ^
TypeError: str.charAt is not a function
    at parse (/home/rombus/Workspace/amberdata/timeseries-kafka-consumer/node_modules/slonik/node_modules/pg-connection-string/index.js:13:10)
    at createPoolConfiguration (/home/rombus/Workspace/amberdata/timeseries-kafka-consumer/node_modules/slonik/src/factories/createPoolConfiguration.js:14:29)
    at Object.createPool (/home/rombus/Workspace/amberdata/timeseries-kafka-consumer/node_modules/slonik/src/factories/createPool.js:33:29)
    at Object.<anonymous> (/home/rombus/Workspace/amberdata/timeseries-kafka-consumer/src/index.ts:25:14)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Module.m._compile (/home/rombus/Workspace/amberdata/timeseries-kafka-consumer/node_modules/ts-node/src/index.ts:814:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/home/rombus/Workspace/amberdata/timeseries-kafka-consumer/node_modules/ts-node/src/index.ts:817:12)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)

```

### PR DESCRIPTION
**Pull request by aimee-gm**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

Commits (in order)
1. Prettier ran (https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes)
1. The typings incorrectly allow you to specify an object as connection configuration. The package only accepts a string. (https://github.com/gajus/slonik/blob/master/src/factories/createPool.js#L22)
1. Added missing Error types (https://github.com/gajus/slonik/blob/master/src/errors.js)

Fixes https://github.com/gajus/slonik/issues/157

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## slonik ([<kbd>unpkg</kbd>](https://unpkg.com/browse/slonik@latest/))
was missing the following properties:
1. createSqlTokenSqlFragment
1. isSqlToken
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 787cb55d5557564e8ac501435ae3a7b234f0171d
</p>

File: types/slonik/index.d.ts [hunk 1]
```diff
@@ -425,12 +392,10 @@ export interface InterceptorType {
...
 export function createFieldNameTransformationInterceptor(configuration: {
-    format: string,
-    test?: (field: FieldType) => boolean
-}): InterceptorType;
-export function createQueryNormalizationInterceptor(configuration?: {
-    stripComments?: boolean;
+    format: string;
+    test?: (field: FieldType) => boolean;
 }): InterceptorType;
+export function createQueryNormalizationInterceptor(configuration?: { stripComments?: boolean }): InterceptorType;
```
Comment (mmkal) [inline types/slonik/index.d.ts:398]: Did you mean to change this?

**Review by mmkal — CHANGES_REQUESTED** [review]
Not sure about the `createQueryNormalizationInterceptor` - might have been deliberate but wanted to check. Also, personally I'd find this easier to review if it were split into several PRs: one for the `createPool` issue, one to add errors, and a formatting-only only one. Having all the whitespace/semi-colon changes mixed in makes it harder to see the important changes.

File: types/slonik/index.d.ts (hunk 1 above)
Comment (aimee-gm) [reply types/slonik/index.d.ts:398]: This is only a formatting change from running prettier (something my editor does automatically (and will cause issues with any PR in the future where the contributor's editor also does this seeing as it's configured), so I think it's worth making the change.

**Comment by aimee-gm** [comment]
@mmkal We may not need these to be seperate PRs; but to help, I've split the changes into 3 commits, which address: formatting, `createPool()` and the errors individually.

**Review by mmkal — APPROVED** [review]
Lgtm - thanks for splitting the changes up!

**Comment by aimee-gm** [comment]
Ready to merge

**Pull request merged**